subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultBuildControllers.java

```diff
@@ -142,7 +142,7 @@ private Comparator<BuildIdentifier> idComparator() {
             if (id2.equals(DefaultBuildIdentifier.ROOT)) {
                 return -1;
             }
-            return id1.getName().compareTo(id2.getName());
+            return id1.getBuildPath().compareTo(id2.getBuildPath());
```

Comment:
That's a good point. I was trying to reduce the surface where the deprecation warning is issued. Because users wouldn't be able to do anything with this warning until Gradle 9.0, which to me seems extremely frustrating.

Do you think we should introduce a method on `DefaultBuildIdentifier` that allows getting the name without triggering the nagging and cast `BuildIdentifier` everywhere we are forced to keep using `getName()` for compatibility reasons?

Reply:
I forgot that we have a mechanism to temporarily silence the deprecation warning if required. But it's not even important, because here we explicitly preserve the order where the root build is always last, so we don't need to revert it to `getName()`.